zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/incident/IncidentResolveProcessor.java

```diff
@@ -72,10 +74,17 @@ public void processRecord(final TypedRecord<IncidentRecord> command) {
       return;
     }
 
+    final long jobKey = incident.getJobKey();
+    if (isJobRelatedIncident(jobKey) && jobState.getJob(jobKey).getRetries() == 0) {
```

Comment:
because of https://github.com/camunda/zeebe/issues/15437 it might be best to check for `<= 0`